test/e2e/framework/node/wait.go

```diff
@@ -116,19 +117,14 @@ func WaitForTotalHealthy(ctx context.Context, c clientset.Interface, timeout tim
 // other than ConditionTrue (e.g. not true or unknown).
 func WaitConditionToBe(ctx context.Context, c clientset.Interface, name string, conditionType v1.NodeConditionType, wantTrue bool, timeout time.Duration) bool {
 	framework.Logf("Waiting up to %v for node %s condition %s to be %t", timeout, name, conditionType, wantTrue)
-	for start := time.Now(); time.Since(start) < timeout; time.Sleep(poll) {
+	return gomega.Eventually(ctx, func() (bool, error) {
 		node, err := c.CoreV1().Nodes().Get(ctx, name, metav1.GetOptions{})
 		if err != nil {
 			framework.Logf("Couldn't get node %s", name)
-			continue
-		}
-
-		if IsConditionSetAsExpected(node, conditionType, wantTrue) {
-			return true
+			return false, err
 		}
-	}
-	framework.Logf("Node %s didn't reach desired %s condition status (%t) within %v", name, conditionType, wantTrue, timeout)
-	return false
+		return IsConditionSetAsExpected(node, conditionType, wantTrue), nil
```

Comment:
Go helper functions which return `bool, err` are not very informative because the `false, nil` case comes with no explanation that could be printed in a progress report.

But you are using some existing function here, so it wouldn't be fair to ask you to improve that.

Reply:
Reverted because the impact is too wide.